lib/api.js

```diff
@@ -0,0 +1,89 @@
+require('dotenv').config();
+const createHttpUserAgent = require('@commercetools/http-user-agent');
+const createHttpClient = require('@octokit/rest');
+const pkg = require('../package.json');
+
+const userAgent = createHttpUserAgent({
+  name: '@octokit/rest',
+  version: pkg.dependencies['@octokit/rest'],
+  libaryName: pkg.name,
+  libaryVersion: pkg.version,
+});
+
+const octokit = createHttpClient({
+  auth: `token ${process.env.AUTH_TOKEN}`,
+  accept: 'application/vnd.github.v3+json',
+  'user-agent': userAgent,
+  // For using descriptions in requests
+  previews: ['symmetra'],
+});
+
+const owner = process.env.OWNER;
+const repo = process.env.REPOSITORY;
+
+async function fetchLabels() {
+  const res = await octokit.issues.listLabelsForRepo({
+    owner,
+    repo,
+  });
+  const labelData = res.data;
+  const formattedLabels = labelData.map(label => {
+    const data = {
+      id: label.id,
+      name: label.name,
+      color: label.color,
+      description: label.description || '',
+    };
+    return data;
+  });
+  return formattedLabels;
+}
+
+function createLabels(labels) {
+  const promiseList = labels.map(label => {
+    return octokit.issues.createLabel({
+      owner,
+      repo,
+      name: label.name,
+      color: label.color,
+      description: label.description || '',
+    });
+  });
+  return Promise.all(promiseList);
+}
+
+function deleteLabels(labels) {
+  const promiseList = labels.map(label => {
+    return octokit.issues.deleteLabel({
+      owner,
+      repo,
+      name: label.name,
+    });
+  });
+  return Promise.all(promiseList);
+}
+
+async function updateLabels(labels) {
+  const oldLabels = await fetchLabels();
+  const promiseList = labels.map(label => {
+    const requiredLabel = oldLabels.find(oldLabel => {
```

Comment:
Careful, `requiredLabel` can be `undefined` if it's not found and then you can't access the properties (name, color, description) below.

Or is it ensured to always be defined?

Reply:
`updateLabels` is only called if the cli knows that some labels were modified. So it should be ensured that the label already exists, otherwise you couldn't "modify" it.
That's why we should always get the required label 🤔